Source/ChilliSource/Core/State/State.cpp

```diff
@@ -54,6 +55,10 @@ namespace ChilliSource
         
         for(auto& system : m_systems)
         {
+            if(system->IsA<Scene>() && system.get() != m_scenes[0])
```

Comment:
This should be done outside this loop, prior to the state init lifecycle event to ensure that all scenes are available via GetScene() during the event. This also means you don't need to handle the main scene differently. (unless something relies on GetMainScene() in the constructor, but I'd hope not